libbeat/processors/add_process_metadata/add_process_metadata.go

```diff
@@ -93,18 +101,44 @@ func newProcessMetadataProcessorWithProvider(cfg *common.Config, provider proces
 		return nil, errors.Wrapf(err, "fail to unpack the %v configuration", processorName)
 	}
 
-	p := addProcessMetadata{
-		config:   config,
-		provider: provider,
-		log:      log,
-	}
-	if p.mappings, err = config.getMappings(); err != nil {
+	mappings, err := config.getMappings()
+
+	if err != nil {
 		return nil, errors.Wrapf(err, "error unpacking %v.target_fields", processorName)
 	}
 
+	var cidProvider gosigarCidProvider
+
+	// don't use cgroup.ProcessCgroupPaths to save it from doing the work when container id disabled
+	if ok := containsValue(mappings, "container.id"); ok {
+		cidProvider = newCidProvider(config.HostPath, config.CgroupPrefixes, processCgroupPaths)
+	} else {
+		dummyProcessCgroupPaths := func(_ string, pid int) (map[string]string, error) {
+			return map[string]string{}, nil
+		}
+		cidProvider = newCidProvider(config.HostPath, []string{}, dummyProcessCgroupPaths)
```

Comment:
```suggestion
		cidProvider = newCidProvider(config.HostPath, nil, dummyProcessCgroupPaths)
```

https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices